tests/test_visitors/test_ast/test_functions/test_getter_without_return.py

```diff
@@ -0,0 +1,249 @@
+class Foo:
+    def get_foo():
+        ...
+"""
+
+getter_method_with_bare_return = """
+class Foo:
+    def get_foo():
+        return
+"""
+
+getter_method_with_valued_return = """
+class Foo:
+    def get_foo():
+        return 1
+"""
+
+getter_method_with_explicit_none_return = """
+class Foo:
+    def get_foo():
+        return None
+"""
+
+getter_method_with_bare_yield = """
+class Foo:
+    def get_foo():
+        yield
+"""
+
+getter_method_with_valued_yield = """
+class Foo:
+    def get_foo():
+        yield 1
+"""
+
+getter_method_with_explicit_none_yield = """
+class Foo:
+    def get_foo():
+        yield None
+"""
+
+getter_method_with_yield_from = """
+class Foo:
+    def get_foo():
+        yield from [1]
+"""
+
+regular_function_with_bare_return = """
+class Foo:
+    def foo():
+        return
+"""
+
+regular_function_with_implicit_return = """
+class Foo:
+    def foo():
+        ...
+"""
+
+regular_function_with_bare_return = """
+class Foo:
+    def foo():
+        return
+"""
+
+regular_function_with_bare_yield = """
+class Foo:
+    def foo():
+        yield
+"""
+
+regular_method_with_bare_return = """
+class Foo:
+    def foo():
+        return
+"""
+
+regular_method_with_implicit_return = """
+class Foo:
+    def foo():
+        ...
+"""
+
+regular_method_with_bare_return = """
+class Foo:
+    def foo():
+        return
+"""
+
+regular_method_with_bare_yield = """
+class Foo:
+    def foo():
+        yield
+"""
+
+getter_method_with_branched_return = """
+def get_foo():
+    if bar:
+        return 1
+"""
+
+
+@pytest.mark.parametrize('code', [
+    getter_function_with_implicit_return,
+    getter_function_with_bare_return,
+    getter_function_with_bare_yield,
+    getter_method_with_implicit_return,
+    getter_method_with_bare_return,
+    getter_method_with_bare_yield,
+])
+def test_wrong_getters(
+    assert_errors,
+    parse_ast_tree,
+    default_options,
+    code,
+    mode,
+):
+    """Testing that getters which do not output values are forbidden."""
+    tree = parse_ast_tree(mode(code))
+
+    visitor = FunctionSignatureVisitor(default_options, tree=tree)
+    visitor.run()
+
+    assert_errors(visitor, [GetterWithoutReturnViolation])
+
+
+@pytest.mark.parametrize('code', [
+    getter_function_with_valued_return,
+    getter_function_with_explicit_none_return,
+    getter_function_with_valued_yield,
+    getter_function_with_explicit_none_yield,
+    getter_method_with_valued_return,
+    getter_method_with_explicit_none_return,
+    getter_method_with_valued_yield,
+    getter_method_with_explicit_none_yield,
+    getter_method_with_branched_return,
+])
+def test_correct_getters(
+    assert_errors,
+    parse_ast_tree,
+    default_options,
+    code,
+    mode,
+):
+    """Testing that getters which output values are allowed."""
+    tree = parse_ast_tree(mode(code))
+
+    visitor = FunctionSignatureVisitor(default_options, tree=tree)
+    visitor.run()
+
+    assert_errors(visitor, [])
```

Comment:
Let's also test a function / method with just a docstring.
It should be allowed: intreface definitions, etc.

We should also allow `get_` function with just `raise` (+ an optional docstring). It is used in interfaces as well.

Reply:
Do we also allow if the `get_` function has extra statements besides the raise statement?
```
def get_foo():
    print('Debug: get_foo is called')
    raise ValueError('error!')
```

Reply:
Nope, only a single `raise`